feat!: Image Embeddings (#101)
Anush008/fastembed-rs: README.md

- [**prithivida/Splade_PP_en_v1**](https://huggingface.co/prithivida/Splade_PP_en_v1) - Default

> ### Image Embedding

Reviewer: Not that `TokenizerFiles` is removed, this [README.md section](https://github.com/Anush008/fastembed-rs#:~:text=results)%3B-,Alternatively%2C%20raw,-.onnx%20files%20can) needs to be updated since it's much simpler now.

```

### Image Embedding

```rust
use fastembed::{ImageEmbedding, ImageInitOptions, ImageEmbeddingModel};

// With default InitOptions
let model = ImageEmbedding::try_new(Default::default())?;

// With custom InitOptions
let model = ImageEmbedding::try_new(ImageInitOptions {
    model_name: ImageEmbeddingModel::ClipVitB32,
    show_download_progress: true,
    ..Default::default()
})?;

let images = vec!["assets/image_0.png", "assets/image_1.png"];

// Generate embeddings with the default batch size, 256
let embeddings = model.embed(images, None)?;

> println!("Embeddings length: {}", embeddings.len()); // -> Embeddings length: 4
> println!("Embedding dimension: {}", embeddings[0].len()); // -> Embedding dimension: 384

Reviewer: ```suggestion
println!("Embeddings length: {}", embeddings.len()); // -> Embeddings length: 2
println!("Embedding dimension: {}", embeddings[0].len()); // -> Embedding dimension: 512
```

println!("Rerank result: {:?}", results);
```

Alternatively, raw `.onnx` files can be loaded through the `UserDefinedEmbeddingModel` struct (for "bring your own" text embedding models) using `TextEmbedding::try_new_from_user_defined(...)`. Similarly,

> "bring your own" image embedding models can be loaded useing the `UserDefinedImageEmbeddingModel` struct and `ImageEmbedding::try_new_from_user_defined(...)`,

Reviewer: ```suggestion
"bring your own" image embedding models can be loaded using the `UserDefinedImageEmbeddingModel` struct and `ImageEmbedding::try_new_from_user_defined(...)`,
```

> Alternatively, raw `.onnx` files can be loaded through the `UserDefinedEmbeddingModel` struct (for "bring your own" text embedding models) using `TextEmbedding::try_new_from_user_defined(...)`. Similarly,
> "bring your own" image embedding models can be loaded using the `UserDefinedImageEmbeddingModel` struct and `ImageEmbedding::try_new_from_user_defined(...)`,
> "bring your own" reranking models can be loaded using the `UserDefinedRerankingModel` struct and `TextRerank::try_new_from_user_defined(...)`.

Reviewer: ```suggestion
Alternatively, local model files can be used for inference via the `try_new_from_user_defined(...)` methods of respective structs.
```

For example:

```rust

> macro_rules! local_model {

Reviewer: Maybe remove the macro? Since it's simple now.

Author: The ·local_model· macro only provides a way to write it and is not included in the crate. I think it can be retained.

Reviewer: Right. I meant to simplify the README.

Reviewer: cc @lywa1998

Reviewer: [`95ef48e` (#101)](https://github.com/Anush008/fastembed-rs/pull/101/commits/95ef48ec5f40e27fc7744ad642e19da68a186542)
